File: samples/da-SnowWizard/SnowWizard/src/services/snow_profiles.ts
//Uses the ServiceNow Rest API to deal with profile information
//Author: crisag@microsoft.com

import axios from 'axios';
import * as dotenv from 'dotenv';

dotenv.config({ path: 'env/.env.local.user' });

class ProfilesApiService {
    
    private SN_INSTANCE: string;
    private SN_USERNAME: string;
    private SN_PASSWORD: string;

    constructor() {
        // Environment variables setup
        this.SN_INSTANCE = process.env.SN_INSTANCE || '';
        this.SN_USERNAME = process.env.SN_USERNAME || '';
        this.SN_PASSWORD = process.env.SN_PASSWORD || '';
    }

    // Function to fetch the latest 10 incidents from ServiceNow
    async getProfile(email: string) {
    try {
        const response = await axios.get(
        `https://${this.SN_INSTANCE}.service-now.com/api/now/table/sys_user`,
        {
            params: {
            sysparm_limit: 10,
            sysparm_query: `email=${email}`
            },
            auth: {
            username: this.SN_USERNAME,
            password: this.SN_PASSWORD
            },
            headers: {
            'Content-Type': 'application/json',
            },
        }
        );
        console.log('Profile fetched successfully from ServiceNow:', response.data.result);
        return response.data.result;
    } catch (error) {
        console.error('Error fetching profile:', error);
        throw error;
    }
    }


}

export default new ProfilesApiService();

File: samples/da-SnowWizard-python/services/snow_profiles.py
# Uses the ServiceNow Rest API to deal with profile information

import os
import requests
from dotenv import load_dotenv

load_dotenv("env/.env.local.user")


class ProfilesApiService:

    def __init__(self):
        self.sn_instance = os.environ.get("SN_INSTANCE", "")
        self.sn_username = os.environ.get("SN_USERNAME", "")
        self.sn_password = os.environ.get("SN_PASSWORD", "")

    @property
    def _base_url(self) -> str:
        return f"https://{self.sn_instance}.service-now.com/api/now/table"

    @property
    def _auth(self) -> tuple:
        return (self.sn_username, self.sn_password)

    @property
    def _headers(self) -> dict:
        return {"Content-Type": "application/json"}

    def get_profile(self, email: str) -> list:
        """Fetch a user profile from ServiceNow by email."""
        response = requests.get(
            f"{self._base_url}/sys_user",
            params={
                "sysparm_limit": 10,
                "sysparm_query": f"email={email}",
            },
            auth=self._auth,
            headers=self._headers,
        )
        response.raise_for_status()
        result = response.json().get("result", [])
        print(f"Profile fetched successfully from ServiceNow: {result}")
        return result
